elephant/kernels.py

```diff
@@ -0,0 +1,339 @@
+
+    :math:`K(t) = \left\{\begin{array}{ll} \frac{1}{2 \tau}, & |t| < \tau \\
+    0, & |t| \geq \tau \end{array} \right`
+    with :math:`\tau = \sqrt{3} \sigma` corresponding to the half width
+    of the kernel.
+
+    Besides the standard deviation `sigma`, for consistency of interfaces the
+    parameter `direction` needed for asymmetric kernels also exists without
+    having any effect in the case of symmetric kernels.
+    """
+    min_cutoff = np.sqrt(3.0)
+
+    def _evaluate(self, t):
+        return (0.5 / (np.sqrt(3.0) * self._sigma_scaled)) * \
+               (np.absolute(t) < np.sqrt(3.0) * self._sigma_scaled)
+
+    def boundary_enclosing_at_least(self,
+                                    fraction=default_kernel_area_fraction):
+        return np.sqrt(3.0) * self.sigma
+
+
+class TriangularKernel(SymmetricKernel):
+    """
+    Class for triangular kernels
+
+    :math:`K(t) = \left\{ \begin{array}{ll} \frac{1}{\tau} (1
+    - \frac{|t|}{\tau}), & |t| < \tau \\ 0, & |t| \geq \tau \end{array} \right`
+    with :math:`\tau = \sqrt{6} \sigma` corresponding to the half width of the
+    kernel.
+
+    Besides the standard deviation `sigma`, for consistency of interfaces the
+    parameter `direction` needed for asymmetric kernels also exists without
+    having any effect in the case of symmetric kernels.
+    """
+    min_cutoff = np.sqrt(6.0)
+
+    def _evaluate(self, t):
+        return (1.0 / (np.sqrt(6.0) * self._sigma_scaled)) * np.maximum(
+            0.0,
+            (1.0 - (np.absolute(t) /
+                    (np.sqrt(6.0) * self._sigma_scaled)).magnitude))
+
+    def boundary_enclosing_at_least(self,
+                                    fraction=default_kernel_area_fraction):
+        return np.sqrt(6.0) * self.sigma
+
+
+class EpanechnikovLikeKernel(SymmetricKernel):
+    """
+    Class for epanechnikov-like kernels
+
+    :math:`K(t) = \left\{\begin{array}{ll} (3 /(4 d)) (1 - (t / d)^2),
+    & |t| < d \\
+    0, & |t| \geq d \end{array} \right`
+    with :math:`d = \sqrt{5} \sigma` being the half width of the kernel.
+
+    The Epanechnikov kernel under full consideration of its axioms has a half
+    width of :math:`sqrt{5}`. Ignoring one axiom also the respective kernel
+    with half width = 1 can be called Epanechnikov kernel.
+    ( https://de.wikipedia.org/wiki/Epanechnikov-Kern )
+    However, arbitrary width of this type of kernel is here preferred to be
+    called 'Epanechnikov-like' kernel.
+
+    Besides the standard deviation `sigma`, for consistency of interfaces the
+    parameter `direction` needed for asymmetric kernels also exists without
+    having any effect in the case of symmetric kernels.
+    """
+    min_cutoff = np.sqrt(5.0)
+
+    def _evaluate(self, t):
+        return (3.0 / (4.0 * np.sqrt(5.0) * self._sigma_scaled)) * np.maximum(
+            0.0,
+            1 - (t / (np.sqrt(5.0) * self._sigma_scaled)).magnitude ** 2)
+
+    def boundary_enclosing_at_least(self,
```

Comment:
Decorator inherit_docstring introduced, from which the methods boundary_enclosing_area_fraction and _evaluate now have documentation inherited from the parent class and optionally further kernel-specific documentation.